Tests/ApolloCodegenTests/CodeGeneration/Templates/EnumTemplateTests.swift

```diff
@@ -484,4 +489,36 @@ class EnumTemplateTests: XCTestCase {
     }
   }
   
+  // MARK: - Schema Customization Tests
+  
+  func test__render__givenEnumAndValues_withCustomNames_shouldRenderWithCustomNames() throws {
+    // given
+    buildSubject(
+      name: "MyEnum",
+      customName: "MyCustomEnum",
+      values: [
+        // Mixed case
+        ("caseOne", nil, nil, nil),
+        ("myCase", nil, nil, "myCustomCase")
+      ]
+    )
+
+    let expected = """
+    \(subject.graphqlEnum.name.typeNameDocumentation)
+    enum MyCustomEnum: String, EnumType {
+      case caseOne = "caseOne"
+      // This type has been renamed from the schema type 'myCase'
+      // using schema customization configuration.
+      case myCustomCase = "myCase"
```

Comment:
I don't know if "using schema customization configuration" is going to be clear to users. I'm also not sure if it's necessary that we actually explain _why_ it's renamed, just that it has been. What do you think?

What about:

```suggestion
      // Renamed from GraphQL schema value: 'myCase'
      case myCustomCase = "myCase"
```